app/experimenter/docs/adrs/0004-monitoring-data-import.md

```diff
@@ -0,0 +1,67 @@
+
+Currently, [Grafana](https://grafana.telemetry.mozilla.org) is used to display monitoring data. It works by querying live tables from BigQuery on-demand and does not have any caching. We would like to eventually replace Grafana capabilities with custom visualizations inside of Experimenter. The goal is to have more flexibility than Grafana gives for how to display the data as well as less overhead for users when context switching from one visualization platform to another.
+
+## Decision Drivers
+
+* Data needs to be displayed "live", updating within minutes
+* Cost needs to stay low
+* Have flexibility with how the data is visualized
+* Data loading speed
+
+
+## Considered Options
+
+1. Add ETL in [biguqery_etl](https://github.com/mozilla/bigquery-etl) to process the live data periodically and output results to a gcs bucket that can be accessed by Experimenter. This is how Jetstream data is currently being accessed.
+2. Add a background celery task that fetches data on-demand from bigquery and stores it in Redis. Redis cache can be used for the next fetch of the same data, but it gets invalidated periodically.
+3. Embed existing Grafana graphs (e.g. [graphs](https://grafana.telemetry.mozilla.org/d/XspgvdxZz/experiment-enrollment?orgId=1&var-experiment_id=bug-1656561-rapid-testing-demo-nightly-80&from=1596153600000&to=1598832000000))
+
+
+## Decision Outcome
+
+##### Chosen option: "(2) Query Data On-Demand from Experimenter"
+
+Option (3) has too many cons compared to (1) and (2) so it can be dismissed. Option (1) is fast but will be quite expensive compared to Option (2). Option (2) would be slower but it is the cheapest of the 3. Although Option (2) is slow, it would not be any slower than Grafana currently is for loading data. In some cases when the cache is hit, it would be faster than Grafana.
+
+Furthermore, in the process of building out monitoring in Experimenter, we can choose to add a hybrid approach down the line. Data that is moved over from Grafana that does not need to be available on-demand can be processed in a less-frequent ETL job.
+
+For this first pass at adding monitoring data, only basic user counts per branch and experiment health metrics will be added.
+
+
+## Pros and Cons of the Options
+
+#### (1) Periodic ETL Output in GCS bucket
+
+* Pros:
+  * No additional complexity for data processing in Experimenter
+  * Allows flexibility for how data is visualized in Experimenter
+  * Data will load faster coming from a bucket than needing to wait for a BigQuery query to run
+* Cons
+  * Since, so far, the demand on Grafana is low, this option to run the full ETL queries every 10 minutes or so may be much more expensive than option (2) or (3) 
+
+#### (2) Query Data On-Demand from Experimenter
+
+* Pros:
+  * It won't be any slower than the current Grafana graphs
+  * Allows flexibility for how data is visualized in Experimenter
+  * The caching layer would make this approach the same price or cheaper than (1) and (2)
```

Comment:
Should this say "(1) and (3)"? (Since this is option (2))

Reply:
yes, thank you